### ISSUE TITLE
Make encoding package available externally

### ISSUE THREAD

**Issue by robdefeo** [mailchain/mailchain#439]
**Is your feature request related to a problem? Please describe.**
The `encoding` package is internal only. It is now almost ready to be exposed.

**Describe the solution you'd like**
- [ ] rename `TypeXXX` to `KindXXX`
```
const (
	// TypeHex encoding value.
	TypeHex = "hex/plain"
	// TypeHex0XPrefix encoding value.
	TypeHex0XPrefix = "hex/0x-prefix"
	// TypeBase58 encoding value.
	TypeBase58 = "base58/plain"
	// TypeBase58SubstrateAddress encoding value.
	TypeBase58SubstrateAddress = "base58/ss58-address"
)
```
- [ ] move `internal/encoding` to `encoding`
- [ ] ensure that all tests pass and binary builds

**Comment by u5surf** [comment]
Hi, Can I help you this refactoring?

**Comment by robdefeo** [comment]
This has been resolved by #440. Thanks for the offer @u5surf 😄
